### PR TITLE
Add 'rails.db.name' tag to ActiveRecord

### PR DESCRIPTION
**Pull request by delner**
This pull request adds the database name as a tag (`rails.db.name`) to ActiveRecord traces.

### CODE REVIEW

File: test/contrib/rails/database_test.rb [hunk 1]
```diff
@@ -1,4 +1,5 @@
 require 'helper'
+require 'pry'
```
Comment (palazzem) [inline test/contrib/rails/database_test.rb]: I think we should remove it, right?

Comment (delner) [reply test/contrib/rails/database_test.rb]: Oh whoops, totally.

Comment (palazzem) [reply test/contrib/rails/database_test.rb]: No worries, reviews are meant for that 💯

File: lib/ddtrace/contrib/rails/utils.rb [hunk 2]
```diff
@@ -50,6 +50,10 @@ def self.adapter_name
           normalize_vendor(connection_adapter)
         end
 
+        def self.database_name
```
Comment (p-lambert) [inline lib/ddtrace/contrib/rails/utils.rb]: considering this method is just an alias, is there any reason for this additional level of indirection?

File: lib/ddtrace/contrib/rails/active_record.rb [hunk 3]
```diff
@@ -21,6 +21,7 @@ def self.sql(_name, start, finish, _id, payload)
           tracer = Datadog.configuration[:rails][:tracer]
           database_service = Datadog.configuration[:rails][:database_service]
           adapter_name = Datadog::Contrib::Rails::Utils.adapter_name
+          database_name = Datadog::Contrib::Rails::Utils.database_name
```
Comment (p-lambert) [inline lib/ddtrace/contrib/rails/active_record.rb]: 👍

File: lib/ddtrace/contrib/rails/utils.rb (hunk 2 above)
Comment (delner) [reply lib/ddtrace/contrib/rails/utils.rb]: I don't think there's much reason beyond just having naming and structure consistent with `adapter_name`. We could reduce this method to an alias or simply remove it altogether.

**Comment by senny** [comment]
@palazzem I hoped this would help me to implement #238. Unfortunately, the tag does not really respect the query it's annotating, it always defaults to the default connection. Any plans to make this work with multiple database connections in the same app?

**Comment by palazzem** [comment]
Hello @senny ! I see, let us investigate on it since the meaning of the tag was just to provide the database name and not exactly the database connection. This should be implemented someway, since you may want to separate properly (for instance) replicas queries from the main database connection.

Is that correct?

**Comment by senny** [comment]
@palazzem yes, ideally we would get the connection name as a tag. In our case though, we are actually using different databases with the same Rails application. What we would need is a way to separate queries that are going to e.g. a reporting or a archive database and not the primary. Since they all have a different database-name, it would have been sufficient for me to implement the split.

We do use follower databases as well where obviously the database name would not be sufficient. In that case we would need the connection name to tell them apart.

**Pull request merged**